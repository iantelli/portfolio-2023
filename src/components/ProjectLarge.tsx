import { motion } from "framer-motion";
import Icon from "./Icon";
import { RiGithubFill, RiLink } from "react-icons/ri";

type ProjectLargeProps = {
  title: string;
  description: string;
  image: string;
  technologies: string[];
  showIcons?: boolean;
  github?: string;
  website?: string;
  pictureLink?: string;
  reverse?: boolean;
};

export default function ProjectLarge({
  title,
  description,
  image,
  technologies,
  github,
  website,
  showIcons = true,
  pictureLink,
  reverse,
}: ProjectLargeProps) {
  return (
    <>
      {reverse ? (
        <motion.div
          className={
            "flex flex-wrap-reverse lg:flex-nowrap justify-between items-center lg:mx-24 lg:my-20 lg:mt-24"
          }
          initial={{ opacity: 0, y: 50 }}
          whileInView={{ opacity: 1, y: 0 }}
          viewport={{ once: true }}
          transition={{ duration: 0.5 }}
        >
          <div className="flex flex-col justify-center items-center text-center lg:justify-start lg:items-start lg:text-start">
            <h1 className="text-2xl lg:text-4xl font-bold font-display mb-4">{title}</h1>
            <h2 className="text-lg lg:text-xl font-semibold text-rosePineDawn-subtle dark:text-rosePine-subtle font-display">
              {description}
            </h2>
            <div className="flex flex-col font-display text-rosePineDawn-subtle dark:text-rosePine-subtle mt-4">
              <div className="flex flex-wrap justify-center lg:justify-start">
                {technologies.map((technology) => (
                  <div
                    key={technology}
                    className="bg-rosePineDawn-highlightMed dark:bg-rosePine-highlightLow text-rosePineDawn-love dark:text-rosePine-foam font-semibold rounded-lg px-2 py-1 mr-2 mb-2"
                  >
                    {technology}
                  </div>
                ))}
              </div>
            </div>
            {showIcons && (
              <div className="flex items-center justify-center lg:items-start lg:justify-start w-full h-full mt-4">
                <Icon
                  newTab
                  hover
                  icon={
                    <RiGithubFill className="h-8 w-8 text-rosePineDawn-love dark:text-rosePine-foam mr-2" />
                  }
                  aria={`link to ${title} github`}
                  link={github!}
                />
                <Icon
                  newTab
                  hover
                  icon={
                    <RiLink className="h-8 w-8 text-rosePineDawn-love dark:text-rosePine-foam" />
                  }
                  aria={`link to ${title} live website`}
                  link={website!}
                />
              </div>
            )}
          </div>
          <a
            className={
              "flex flex-row justify-center items-center lg:justify-center lg:items-center lg:text-start w-full h-full mb-4 lg:mb-0 lg:ml-4"
            }
            href={pictureLink ? pictureLink : github}
            target="_blank"
            rel="noreferrer"
          >
            <motion.img
              src={image}
              alt={title}
              className="lg:grayscale lg:hover:grayscale-0 lg:transition lg:duration-200 min-w-40 md:w-full rounded-lg shadow-lg"
              initial={{ scale: 1 }}
              whileHover={{ scale: 1.05, transition: { duration: 0.1 } }}
            />
          </a>
        </motion.div>
      ) : (
        <motion.div
          className={
            "flex flex-wrap lg:flex-nowrap justify-between items-center lg:mx-24 lg:my-20 lg:mt-24"
          }
          initial={{ opacity: 0, y: 50 }}
          whileInView={{ opacity: 1, y: 0 }}
          viewport={{ once: true }}
          transition={{ duration: 0.5 }}
        >
          <a
            className={
              "flex flex-row justify-center items-center lg:justify-center lg:items-center lg:text-start w-full h-full mb-4 lg:mb-0 lg:mr-4"
            }
            href={pictureLink ? pictureLink : github}
            target="_blank"
            rel="noreferrer"
          >
            <motion.img
              src={image}
              alt={title}
              className="lg:grayscale lg:hover:grayscale-0 lg:transition min-w-40 md:w-full lg:duration-200 rounded-lg shadow-lg"
              initial={{ scale: 1 }}
              whileHover={{ scale: 1.05, transition: { duration: 0.1 } }}
            />
          </a>
          <div className="flex flex-col justify-center items-center text-center lg:justify-center lg:items-end lg:text-end">
            <h1 className="text-2xl lg:text-4xl font-bold font-display mb-4">{title}</h1>
            <h2 className="text-lg lg:text-xl font-semibold text-rosePineDawn-subtle dark:text-rosePine-subtle font-display">
              {description}
            </h2>
            <div
              className="flex flex-col font-display text-rosePineDawn-subtle dark:text-rosePine-subtle mt-4">
              <div className="flex flex-wrap justify-center lg:justify-end">
                {technologies.map((technology) => (
                  <div
                    key={technology}
                    className="bg-rosePineDawn-highlightMed dark:bg-rosePine-highlightLow text-rosePineDawn-love dark:text-rosePine-foam font-semibold rounded-lg px-2 py-1 mr-2 mb-2"
                  >
                    {technology}
                  </div>
                ))}
              </div>
            </div>
            {showIcons && (
              <div className="flex items-center justify-center lg:items-start lg:justify-end w-full h-full mt-4">
                <Icon
                  newTab
                  hover
                  icon={
                    <RiGithubFill className="h-8 w-8 text-rosePineDawn-love dark:text-rosePine-foam" />
                  }
                  aria={`link to ${title} github`}
                  link={github!}
                />
                <Icon
                  newTab
                  hover
                  icon={
                    <RiLink className="h-8 w-8 text-rosePineDawn-love dark:text-rosePine-foam ml-2" />
                  }
                  aria={`link to ${title} website`}
                  link={website!}
                />
              </div>
            )}
          </div>
        </motion.div>
      )}
    </>
  );
}
