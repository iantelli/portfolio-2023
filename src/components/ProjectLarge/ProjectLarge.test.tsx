import { render } from "@testing-library/react";
import { expect, it } from "vitest";
import ProjectLarge from ".";

describe("ProjectLarge", () => {
  const props = {
    title: "Discord Bot",
    description: "A Discord bot that does stuff",
    image: "InterviewPal.webp",
    technologies: ["React", "TypeScript", "TailwindCSS"],
  };
  it("renders ProjectLarge component", () => {
    const component = render(<ProjectLarge title={props.title} description={props.description} image={props.image} technologies={props.technologies}/>);
    expect(component).toMatchSnapshot();
  });
});
