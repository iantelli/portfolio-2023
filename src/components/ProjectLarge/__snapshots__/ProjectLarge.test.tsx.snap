// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProjectLarge > renders ProjectLarge component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex flex-wrap lg:flex-nowrap justify-between items-center lg:mx-24 lg:my-20 lg:mt-24"
        style="opacity: 0; transform: translateY(50px) translateZ(0);"
      >
        <a
          class="flex flex-row justify-center items-center lg:justify-center lg:items-center lg:text-start w-full h-full mb-4 lg:mb-0 lg:mr-4"
          rel="noreferrer"
          target="_blank"
        >
          <img
            alt="Discord Bot"
            class="lg:grayscale lg:hover:grayscale-0 lg:transition min-w-40 md:w-full lg:duration-200 rounded-lg shadow-lg"
            src="InterviewPal.webp"
            style="transform: none;"
          />
        </a>
        <div
          class="flex flex-col justify-center items-center text-center lg:justify-center lg:items-end lg:text-end"
        >
          <h1
            class="text-2xl lg:text-4xl font-bold font-display mb-4"
          >
            Discord Bot
          </h1>
          <h2
            class="text-lg lg:text-xl font-semibold text-rosePineDawn-subtle dark:text-rosePine-subtle font-display"
          >
            A Discord bot that does stuff
          </h2>
          <div
            class="flex flex-col font-display text-rosePineDawn-subtle dark:text-rosePine-subtle mt-4"
          >
            <div
              class="flex flex-wrap justify-center lg:justify-end"
            >
              <div
                class="bg-rosePineDawn-highlightMed dark:bg-rosePine-highlightLow text-rosePineDawn-love dark:text-rosePine-foam font-semibold rounded-lg px-2 py-1 mr-2 mb-2"
              >
                React
              </div>
              <div
                class="bg-rosePineDawn-highlightMed dark:bg-rosePine-highlightLow text-rosePineDawn-love dark:text-rosePine-foam font-semibold rounded-lg px-2 py-1 mr-2 mb-2"
              >
                TypeScript
              </div>
              <div
                class="bg-rosePineDawn-highlightMed dark:bg-rosePine-highlightLow text-rosePineDawn-love dark:text-rosePine-foam font-semibold rounded-lg px-2 py-1 mr-2 mb-2"
              >
                TailwindCSS
              </div>
            </div>
          </div>
          <div
            class="flex items-center justify-center lg:items-start lg:justify-end w-full h-full mt-4"
          >
            <a
              aria-label="link to Discord Bot github"
              rel="noreferrer"
              target="_blank"
            >
              <div
                class="flex items-center justify-center h-auto w-auto"
              >
                <div>
                  <svg
                    class="h-8 w-8 text-rosePineDawn-love dark:text-rosePine-foam"
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <g>
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                      />
                      <path
                        d="M12 2C6.475 2 2 6.475 2 12a9.994 9.994 0 0 0 6.838 9.488c.5.087.687-.213.687-.476 0-.237-.013-1.024-.013-1.862-2.512.463-3.162-.612-3.362-1.175-.113-.288-.6-1.175-1.025-1.413-.35-.187-.85-.65-.013-.662.788-.013 1.35.725 1.538 1.025.9 1.512 2.338 1.087 2.912.825.088-.65.35-1.087.638-1.337-2.225-.25-4.55-1.113-4.55-4.938 0-1.088.387-1.987 1.025-2.688-.1-.25-.45-1.275.1-2.65 0 0 .837-.262 2.75 1.026a9.28 9.28 0 0 1 2.5-.338c.85 0 1.7.112 2.5.337 1.912-1.3 2.75-1.024 2.75-1.024.55 1.375.2 2.4.1 2.65.637.7 1.025 1.587 1.025 2.687 0 3.838-2.337 4.688-4.562 4.938.362.312.675.912.675 1.85 0 1.337-.013 2.412-.013 2.75 0 .262.188.574.688.474A10.016 10.016 0 0 0 22 12c0-5.525-4.475-10-10-10z"
                      />
                    </g>
                  </svg>
                </div>
              </div>
            </a>
            <a
              aria-label="link to Discord Bot website"
              rel="noreferrer"
              target="_blank"
            >
              <div
                class="flex items-center justify-center h-auto w-auto"
              >
                <div>
                  <svg
                    class="h-8 w-8 text-rosePineDawn-love dark:text-rosePine-foam ml-2"
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <g>
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                      />
                      <path
                        d="M18.364 15.536L16.95 14.12l1.414-1.414a5 5 0 1 0-7.071-7.071L9.879 7.05 8.464 5.636 9.88 4.222a7 7 0 0 1 9.9 9.9l-1.415 1.414zm-2.828 2.828l-1.415 1.414a7 7 0 0 1-9.9-9.9l1.415-1.414L7.05 9.88l-1.414 1.414a5 5 0 1 0 7.071 7.071l1.414-1.414 1.415 1.414zm-.708-10.607l1.415 1.415-7.071 7.07-1.415-1.414 7.071-7.07z"
                      />
                    </g>
                  </svg>
                </div>
              </div>
            </a>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex flex-wrap lg:flex-nowrap justify-between items-center lg:mx-24 lg:my-20 lg:mt-24"
      style="opacity: 0; transform: translateY(50px) translateZ(0);"
    >
      <a
        class="flex flex-row justify-center items-center lg:justify-center lg:items-center lg:text-start w-full h-full mb-4 lg:mb-0 lg:mr-4"
        rel="noreferrer"
        target="_blank"
      >
        <img
          alt="Discord Bot"
          class="lg:grayscale lg:hover:grayscale-0 lg:transition min-w-40 md:w-full lg:duration-200 rounded-lg shadow-lg"
          src="InterviewPal.webp"
          style="transform: none;"
        />
      </a>
      <div
        class="flex flex-col justify-center items-center text-center lg:justify-center lg:items-end lg:text-end"
      >
        <h1
          class="text-2xl lg:text-4xl font-bold font-display mb-4"
        >
          Discord Bot
        </h1>
        <h2
          class="text-lg lg:text-xl font-semibold text-rosePineDawn-subtle dark:text-rosePine-subtle font-display"
        >
          A Discord bot that does stuff
        </h2>
        <div
          class="flex flex-col font-display text-rosePineDawn-subtle dark:text-rosePine-subtle mt-4"
        >
          <div
            class="flex flex-wrap justify-center lg:justify-end"
          >
            <div
              class="bg-rosePineDawn-highlightMed dark:bg-rosePine-highlightLow text-rosePineDawn-love dark:text-rosePine-foam font-semibold rounded-lg px-2 py-1 mr-2 mb-2"
            >
              React
            </div>
            <div
              class="bg-rosePineDawn-highlightMed dark:bg-rosePine-highlightLow text-rosePineDawn-love dark:text-rosePine-foam font-semibold rounded-lg px-2 py-1 mr-2 mb-2"
            >
              TypeScript
            </div>
            <div
              class="bg-rosePineDawn-highlightMed dark:bg-rosePine-highlightLow text-rosePineDawn-love dark:text-rosePine-foam font-semibold rounded-lg px-2 py-1 mr-2 mb-2"
            >
              TailwindCSS
            </div>
          </div>
        </div>
        <div
          class="flex items-center justify-center lg:items-start lg:justify-end w-full h-full mt-4"
        >
          <a
            aria-label="link to Discord Bot github"
            rel="noreferrer"
            target="_blank"
          >
            <div
              class="flex items-center justify-center h-auto w-auto"
            >
              <div>
                <svg
                  class="h-8 w-8 text-rosePineDawn-love dark:text-rosePine-foam"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g>
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                    <path
                      d="M12 2C6.475 2 2 6.475 2 12a9.994 9.994 0 0 0 6.838 9.488c.5.087.687-.213.687-.476 0-.237-.013-1.024-.013-1.862-2.512.463-3.162-.612-3.362-1.175-.113-.288-.6-1.175-1.025-1.413-.35-.187-.85-.65-.013-.662.788-.013 1.35.725 1.538 1.025.9 1.512 2.338 1.087 2.912.825.088-.65.35-1.087.638-1.337-2.225-.25-4.55-1.113-4.55-4.938 0-1.088.387-1.987 1.025-2.688-.1-.25-.45-1.275.1-2.65 0 0 .837-.262 2.75 1.026a9.28 9.28 0 0 1 2.5-.338c.85 0 1.7.112 2.5.337 1.912-1.3 2.75-1.024 2.75-1.024.55 1.375.2 2.4.1 2.65.637.7 1.025 1.587 1.025 2.687 0 3.838-2.337 4.688-4.562 4.938.362.312.675.912.675 1.85 0 1.337-.013 2.412-.013 2.75 0 .262.188.574.688.474A10.016 10.016 0 0 0 22 12c0-5.525-4.475-10-10-10z"
                    />
                  </g>
                </svg>
              </div>
            </div>
          </a>
          <a
            aria-label="link to Discord Bot website"
            rel="noreferrer"
            target="_blank"
          >
            <div
              class="flex items-center justify-center h-auto w-auto"
            >
              <div>
                <svg
                  class="h-8 w-8 text-rosePineDawn-love dark:text-rosePine-foam ml-2"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g>
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                    <path
                      d="M18.364 15.536L16.95 14.12l1.414-1.414a5 5 0 1 0-7.071-7.071L9.879 7.05 8.464 5.636 9.88 4.222a7 7 0 0 1 9.9 9.9l-1.415 1.414zm-2.828 2.828l-1.415 1.414a7 7 0 0 1-9.9-9.9l1.415-1.414L7.05 9.88l-1.414 1.414a5 5 0 1 0 7.071 7.071l1.414-1.414 1.415 1.414zm-.708-10.607l1.415 1.415-7.071 7.07-1.415-1.414 7.071-7.07z"
                    />
                  </g>
                </svg>
              </div>
            </div>
          </a>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
